### PR TITLE
[Git] Add pretty-formats mini-language

### PR DESCRIPTION
**Pull request by michaelblyons**
https://git-scm.com/docs/pretty-formats

### CODE REVIEW

File: Git Formats/Git Pretty Formats.sublime-syntax [hunk 1]
```diff
@@ -0,0 +1,153 @@
+%YAML 1.2
+---
+# Syntax based on documentation here:
+# https://git-scm.com/docs/pretty-formats
+
+name: Git Pretty Formats
+hidden: true
+scope: text.git.pretty-formats
```
[the remark this answers is not in the document]
Comment (michaelblyons) [reply Git Formats/Git Pretty Formats.sublime-syntax]: I would be fine with this. Then we could put [`for-each-ref` formats][1] in the same file.

[1]: https://git-scm.com/docs/git-for-each-ref#_field_names

[the remark this answers is not in the document]
Comment (deathaxe) [reply Git Formats/Git Pretty Formats.sublime-syntax]: Yep. The _Git Common.sublime-syntay_ is meant to be the library for generally used contexts along with all the different file types.

We would just need to move all the syntax tests to one of the including syntaxes to make them work properly. Here you'd need to move them to the Git Config.

Including all the different scopes to the _Git Common_ main scope may cause conflicts at some point.

[the remark this answers is not in the document]
Comment (deathaxe) [reply Git Formats/Git Pretty Formats.sublime-syntax]: By adding the `##[ name ]####` comments all the parts are clearly distinguished and can be folded if not worked at.

File: Git Formats/Git Pretty Formats.sublime-syntax [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+        - match: reset
+          scope: support.constant.color-reset.git.config
+
+  trailers:
+    - match: '%(\()(trailers)(\))'
+      captures:
+        1: punctuation.section.parens.begin.git.pretty-formats
+        2: keyword.other.trailers.git.pretty-formats
+        3: punctuation.section.parens.end.git.pretty-formats
+    - match: '%(\()(trailers)(:)'
```
[the remark this answers is not in the document]
Comment (michaelblyons) [reply Git Formats/Git Pretty Formats.sublime-syntax]: Would you really use line continuations with this syntax? I haven't tried, but I imagine that indenting the next line will put a bunch of unwanted spaces into your "pretty" format.

[the remark this answers is not in the document]
Comment (deathaxe) [reply Git Formats/Git Pretty Formats.sublime-syntax]: Don't know which insane ideas some users might have. I would probably not use them. I just find it better to keep patterns as short as possible in order to ...

1. try highlighting incomplete expressions
2. try to reduce the amount of capture groups to improve performance

It just feels better if you start writing an expression and it starts highlighting properly as early as possible as the editor already knew what you are going to write.

[the remark this answers is not in the document]
Comment (deathaxe) [reply Git Formats/Git Pretty Formats.sublime-syntax]: Created a PR at your repo with some (maybe inomplete) ideas.

[the remark this answers is not in the document]
Comment (michaelblyons) [reply Git Formats/Git Pretty Formats.sublime-syntax]: Thanks! I'll take a look.

**Comment by wbond** [comment]
Wow, that documentation page does not do a very good job at giving a high-level overview. 🙂 That whole first paragraph seems like it is an implementation detail that should be mentioned *after* explaining the pretty formats.

**Pull request merged**